### ISSUE TITLE
Real container state lost across reboot

### ISSUE THREAD

**Issue by cjeanner** [containers/libpod#1703]
**Is this a BUG REPORT or FEATURE REQUEST?**:

[//]: # Uncomment only one, leave it on its own line:

# **kind bug**

[//]: # **kind feature**

**Description**

**Steps to reproduce the issue:**

1. Run a container that fails starting properly. It should be listed with an "Exited (non-0)" in ```podman ps --all```

2. Reboot the computer/server

**Describe the results you received:**
After the reboot, ```podman ps --all``` shows the container as "Created"

**Describe the results you expected:**
It should keep the "Exited (error-code)" instead, in order to stay consistent.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
podman version 0.9.4-dev
```

**Output of `podman info`:**

```
host:
  BuildahVersion: 1.4-dev
  Conmon:
    package: podman-0.9.3.1-3.gita3c4ce6.el7.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 1.11.4, commit: 88d5276957861dea88b85f1d3f786612042cb4dd-dirty'
  Distribution:
    distribution: '"centos"'
    version: "7"
  MemFree: 8404660224
  MemTotal: 15871639552
  OCIRuntime:
    package: runc-1.0.0-54.dev.git2abd837.el7.x86_64
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.0'
  SwapFree: 0
  SwapTotal: 0
  arch: amd64
  cpus: 6
  hostname: undercloud.lain.internux.ch
  kernel: 3.10.0-862.14.4.el7.x86_64
  os: linux
  uptime: 10m 3.02s
insecure registries:
  registries:
  - 192.168.24.1:8787
  - 192.168.24.3:8787
registries:
  registries:
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.access.redhat.com
  - registry.centos.org
store:
  ContainerStore:
    number: 90
  GraphDriverName: overlay
  GraphOptions:
  - overlay.override_kernel_check=true
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
  ImageStore:
    number: 37
  RunRoot: /var/run/containers/storage
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Running on a libvirt domain - although this shouldn't affect the state reported by podman.

**Comment by TomSweeneyRedHat** [comment]
@cjeanner thanks for the problem report.  Do you by chance have a Dockerfile or a recommended way to create a container that fails to start?

**Comment by cjeanner** [comment]
Hello @TomasTomecek 
I guess a command as "exit 1" in the Dockerfile should do the trick. I can confirm that in about an hour if you want - I have some running tests I can't really stop right now, and a bit out of resources :/

Cheers,
C.

**Comment by mheon** [comment]
This should be pretty easy now that we have a dedicated Exited state.
Instead of unconditionally setting the Created state, we set it only if we
are not in Exited state. Stopped also probably transitions to Exited on
refresh.

**Comment by rhatdan** [comment]
@mheon Has this been fixed?

**Comment by mheon** [comment]
Nope. I'll take it, should be a quick fix.

**Comment by cjeanner** [comment]
@mheon I guess we can close this issue now since you merged #2273 ?

**Comment by mheon** [comment]
Yes. Good call.
